### ISSUE TITLE
PA API 4.0 shutting down October 31, 2019

### ISSUE THREAD

**Issue by sacapuntas9** [lionheart/bottlenose#78]
Based on this info: https://webservices.amazon.com/paapi5/documentation/faq.html

it seems that the PA API 4.0 is being deprecated very soon; this was just announced today. Will this wrapper work at this time, and if not, what modifications are necessary to keep it working?

**Comment by tinohager** [comment]
For information on this project

[The old Amazon Product Api is shut down on `31 October 2019`](https://dev.to/skatkov/new-amazon-product-advertising-api-is-here-and-you-need-to-migrate-fast-3545)

Documentation for the new API
- [API Reference](https://webservices.amazon.com/paapi5/documentation/)
- [Product Advertising API Scratchpad](https://webservices.amazon.com/paapi5/documentation/play-around-using-scratchpad.html)

**Comment by Cealtea** [comment]
Will this library be updated for the new API???
If not please let us know so we can switch away from it.
Thank you.

**Comment by iranzo** [comment]
Deadline has been moved to 9th March, but still would be good to get information about PAAPI 5

**Comment by Cealtea** [comment]
Amazon has its own SDK for Products API v5 now. 
https://webservices.amazon.com/paapi5/documentation/quick-start/using-sdk.html

**Comment by sergioteula** [comment]
Hello everyone, I have created my own wrapper for the new release of the Amazon API. I hope it will be helpful. https://github.com/sergioteula/python-amazon-paapi5

**Comment by alefiori82** [comment]
Hi everyone.
I make a python wrapper for the new Amazon API. As bottlenose can use cache function to get and write results. You find the package at https://github.com/alefiori82/amazon-paapi5 or you can install from pypi at https://pypi.org/project/amazon-paapi5/

**Comment by alefiori82** [comment]
> Hi everyone.
> I make a python wrapper for the new Amazon API. As bottlenose can use cache function to get and write results. You find the package at https://github.com/alefiori82/amazon-paapi5 or you can install from pypi at https://pypi.org/project/amazon-paapi5/

I will upload a new version of the library in the next days to improve the cache reader and writer and  override the objects returned by the calls. I will try to add also a wikibook for the documentation.

**Comment by carlosevi94** [comment]
It will still works on PA API 5 without updates?

**Comment by sergioteula** [comment]
> It will still works on PA API 5 without updates?

No, it won't. But you can switch to my wrapper, I hope it would be useful.

**Comment by alefiori82** [comment]
The new version (1.1.0) of amazon-paapi5 has been released. You can download or clone the repository from [github](https://github.com/alefiori82/amazon-paapi5 ) or installi it using [pypi](https://pypi.org/project/amazon-paapi5/) 

``pip install amazon-paapi5``

The documentation is available at  [Read the docs](https://amazon-paapi5.readthedocs.io/en/latest/index.html`)
Send me also suggestions for further implementations using the [issue](https://github.com/alefiori82/amazon-paapi5/issues) page.
To support the development please visit the [sponsor](https://www.buymeacoffee.com/1jTO4Av) page.

**Comment by michaelx** [comment]
@dlo Thank you so much for bottlenose, it always worked great! Any plans to make it work with PA API 5? I’d like to avoid migrating back and forth between libraries.

**Comment by BorFour** [comment]
@michaelx It looks to me that bottlenose won't support PA API v5, unfortunately, so the alternatives so far would to:
- Use [Amazon's SDK](https://webservices.amazon.com/paapi5/documentation/quick-start/using-sdk.html). Apparently, you'd have to copy the source code as it seems that it is not available as a pip package
- Use the [amazon-paapi5](https://github.com/alefiori82/amazon-paapi5) wrapper
- Use the [python-amazon-paapi5](https://github.com/sergioteula/python-amazon-paapi5) wrapper

Looking at [this document](https://webservices.amazon.com/paapi5/documentation/migration-guide/pa-api-40-to-50-mapping.html), not all the operations change, so maybe bottlenose keeps working just with those that don't change.

**Comment by carlosevi94** [comment]
> @michaelx It looks to me that bottlenose won't support PA API v5, unfortunately, so the alternatives so far would be to:
> 
>     * Use [Amazon's SDK](https://webservices.amazon.com/paapi5/documentation/quick-start/using-sdk.html). Apparently, you'd have to copy the source code as it seems that it is not available as a pip package
> 
>     * Use the [amazon-paapi5](https://github.com/alefiori82/amazon-paapi5) wrapper
> 
>     * Use the [python-amazon-paapi5](https://github.com/sergioteula/python-amazon-paapi5) wrapper
> 
> 
> Looking at [this document](https://webservices.amazon.com/paapi5/documentation/migration-guide/pa-api-40-to-50-mapping.html), not all the operations change, so maybe bottlenose keeps working just with those that remain the same.

Amazon's SDK is available installing `pip install amightygirl.paapi5-python-sdk`

**Comment by BorFour** [comment]
@carlosevi94 You are right! I saw before but I was confused by "amightygirl", but reading the package description it looks like it is official

**Comment by alefiori82** [comment]
The amightygirl.paapi5-python-sdk is the official sdk of amazon. My [wrapper](https://github.com/alefiori82/amazon-paapi5/) is based on it and you can install with pip. I just simplified some calls, add some additional computed information to the Products returned and added the usage of the cache as in bottlenose.

**Comment by sergioteula** [comment]
My wrapper https://github.com/sergioteula/python-amazon-paapi has been updated with new structure to make it simpler and access to the unparsed data from the API. I hope it helps.

**Comment by sergioteula** [comment]
Now also compatible with search and get variations functionalities. https://github.com/sergioteula/python-amazon-paapi

**Comment by YPCrumble** [comment]
Both of these wrappers have switched the package to GPLv3 licensing - it would be nice if one could revert to Bottlenose's Apache license or move to a permissive license. I and probably a lot of developers can't use GPLv3'd packages due to the strong copyleft provision.

**Comment by sergioteula** [comment]
> Both of these wrappers have switched the package to GPLv3 licensing - it would be nice if one could revert to Bottlenose's Apache license or move to a permissive license. I and probably a lot of developers can't use GPLv3'd packages due to the strong copyleft provision.

Hello, thanks for the suggestion. Already changed to MIT License, I hope it helps.

**Comment by alefiori82** [comment]
Hi. License updated and new release available. As soon as possible the wrapper will be also available for python 2.

**Comment by shubhendusaurabh** [comment]
> You are right! I saw before but I was confused by "amightygirl", but reading the package description it looks like it is official

the code looks official but maintained by unofficial account.
